src/databricks/labs/lsql/dashboards.py

```diff
@@ -51,16 +65,53 @@ def as_dict(self) -> dict[str, str]:
 
 @dataclass
 class WidgetMetadata:
-    order: int
-    width: int
-    height: int
+    path: Path
+    order: int = 0
+    width: int = 0
+    height: int = 0
+    id: str = ""
+
+    def __post_init__(self):
+        width, height = self._size
+        self.width = self.width or width
+        self.height = self.height or height
+        self.id = self.id or self.path.stem
+
+    @property
+    def spec_type(self) -> type[WidgetSpec]:
+        if self.path.suffix == ".md":
+            return MarkdownSpec
+        # TODO: When supporting more specs, infer spec from query
+        return CounterSpec
+
+    @property
+    def _size(self) -> tuple[int, int]:
+        """Get the width and height for a widget.
+
+        The tiling logic works if:
+        - width < _MAXIMUM_DASHBOARD_WIDTH : heights for widgets on the same row should be equal
+        - width == _MAXIMUM_DASHBOARD_WIDTH : any height
+        """
+        if self.spec_type is None:
+            return 0, 0
+        if self.spec_type == CounterSpec:
```

Comment:
This should be overridden in the CounterSpec class, not in this class

Reply:
What do you mean with overriden? There is nothing overriden here